doc/source/reference/sparse.migration_to_sparray.rst

```diff
@@ -0,0 +1,286 @@
+
+-  Second pass (switching to sparray):
+
+   -  Convert construction functions like ``diags`` and ``triu`` to the
+      array version (see :ref:`sparse-migration-construction` below).
+   -  Check all functions/methods for which migration causes 1D return
+      values. These are mostly indexing and the reduction functions
+      (see :ref:`sparse-migration-shapes-reductions` below).
+   -  Check all places where you iterate over spmatrices and change them
+      to account for the sparrays yielding 1D sparrays rather than 2D spmatrices.
+   -  Find and change places where your code makes use of ``np.matrix``
+      features. Convert those to ``np.ndarray`` features.
+   -  Rename all ``*_matrix`` constructor calls to ``*_array``.
+   -  Test your code. And **read** your code. You have migrated to
+      sparray.
+
+.. _sparse-migration-construction:
+
+Details: construction functions
+-------------------------------
+
+These four functions are new and only handle sparrays:
+:func:`~scipy.sparse.block_array`, :func:`~scipy.sparse.diags_array`,
+:func:`~scipy.sparse.eye_array`, and :func:`~scipy.sparse.random_array`.
+Their signatures are::
+
+   def block_array(blocks, format=None, dtype=None):
+   def diags_array(diagonals, /, *, offsets=0, shape=None, format=None, dtype=None):
+   def eye_array(m, n=None, *, k=0, dtype=float, format=None):
+   def random_array(m, n, density=0.01, format='coo', dtype=None, random_state=None, data_random_state=None):
+
+Existing functions you should be careful while migrating
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+These functions return sparray or spmatrix, depending on the input types they
+receive: :func:`~scipy.sparse.kron`, :func:`~scipy.sparse.kronsum`,
+:func:`~scipy.sparse.hstack`, :func:`~scipy.sparse.vstack`,
+:func:`~scipy.sparse.block_diag`, :func:`~scipy.sparse.tril`, and
+:func:`~scipy.sparse.triu`. Their signatures are::
+
+   def kron(A, B, format=None):
+   def kronsum(A, B, format=None):
+   def hstack(blocks, format=None, dtype=None):
+   def vstack(blocks, format=None, dtype=None):
+   def block_diag(mats, format=None, dtype=None):
+   def tril(A, k=0, format=None):
+   def triu(A, k=0, format=None):
+
+Use of these functions should be examined and inputs adjusted to ensure return
+values are sparrays. And in turn the outputs should be treated as sparrays.
+To return sparrays, at least one input must be an sparray. If you use
+list-of-lists or numpy arrays as input you should convert one of them
+to a sparse array to get sparse arrays out.
+
+Functions that changed names for the migration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+
+   +=========+=============+
+   |Function | New function|
+   +=========+=============+
+   |eye      | eye_array   |
+   |identity | eye_array   |
+   |diags    | diags_array |
+   |spdiags  | diags_array |
+   |bmat     | block       |
+   |rand     | random_array|
+   |random   | random_array|
+   +=========+=============+
```

Comment:
These tables don't render particularly nicely, but I'm happy to merge them - the perfect is the enemy of the good here I think.

Reply:
I changed the tables to rst simple table format in the latest commit. I like the ascii format better but this scales with browser size better. Let me know if you prefer to return to the ascii or suggest another way to make it work.